kong/plugins/rate-limiting/handler.lua

```diff
@@ -172,38 +163,22 @@ function RateLimitingHandler:access(conf)
       headers[RATELIMIT_LIMIT] = limit
       headers[RATELIMIT_REMAINING] = remaining
       headers[RATELIMIT_RESET] = reset
-
-      kong.ctx.plugin.headers = headers
     end
 
     -- If limit is exceeded, terminate the request
     if stop then
-      return kong.response.exit(429, { message = "API rate limit exceeded" }, {
-        [RETRY_AFTER] = reset
-      })
+      headers[RETRY_AFTER] = reset
+      return kong.response.exit(429, { message = "API rate limit exceeded" }, headers)
```

Comment:
If I'm understanding this correctly, by reusing the `headers` variable we're sending more headers than before when the rate limit is exceeded. Before it was just `RETRY_AFTER`, now it's that plus any others already in the `headers` var (like `RATELIMIT_LIMIT`).

This sounds like a change (non-breaking, probably, but a change). Is this desirable / accounted for?

Reply:
@kikito we are not sending more, In previous version it sent the the rest in `header_filter`.